llama-index-integrations/indices/llama-index-indices-managed-llama-cloud/llama_index/indices/managed/llama_cloud/retriever.py

```diff
@@ -90,9 +96,19 @@ def _retrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
     async def _aretrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
         """Asynchronously retrieve from the platform."""
         pipelines = await self._aclient.pipeline.search_pipelines(
-            project_name=self.project_name, pipeline_name=self.name
+            project_name=self.project_name,
+            pipeline_name=self.name,
+            pipeline_type=PipelineType.MANAGED,
         )
-        assert len(pipelines) == 1
+        if len(pipelines) == 0:
+            raise ValueError(
+                f"Unknown index name {self.name}. Please confirm a "
+                "managed index with this name exists."
+            )
+        elif len(pipelines) > 1:
+            raise ValueError(
+                f"Multiple pipelines found with name {self.name} in project {self.project_name}"
```

Comment:
This is possible? I though project name +index name had to be unique lol

Reply:
well, without the `pipeline_type` filter, yes it is possible. You can have two pipelines of the same name as long as they're different types (managed vs playground). Now that I've added this `pipeline_type` filter above however, I don't think this `elif len(pipelines) > 1` condition can ever be possible.